File: clojure/difference-of-squares/src/difference_of_squares.clj
(ns difference-of-squares)

(defn- square [n] (* n n))

(defn sum-of-squares [n]
  (->> (inc n)
       (range 1)
       (map square)
       (reduce +)))

(defn square-of-sum [n]
  (->> (inc n)
       (range 1)
       (reduce +)
       square))

(defn difference [n]
  (- (square-of-sum n)
     (sum-of-squares n)))

File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// SquareOfSum calculates the square of sum for a given number
func SquareOfSum(n int) int {
	res := (n + 1) * n / 2
	return res * res
}

// SumOfSquares calculates the sum of squares for a given number
func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

// Difference calculates the difference between square of sum
// and sum of square for a given number
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
